tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -310,39 +312,77 @@ Target GenerateNoopStages
+
+           var profilerExcludePaths = profilerConfig.On?.PullRequest?.PathsIgnore ?? Array.Empty<string>();
+           Matcher profilerPathMatcher = new();
+           profilerPathMatcher.AddInclude("**");
+           profilerPathMatcher.AddExcludePatterns(profilerExcludePaths);
+
+           Logger.Info($"Found {profilerExcludePaths.Length} exclude paths for the profiler");
 
            var gitChanges = GetGitChangedFiles(baseBranch);
            Logger.Info($"Found {gitChanges.Length} modified paths");
 
            var willConsolidatedPipelineRun = gitChanges.Any(
-               changed => !excludePaths.Any(prefix => changed.StartsWith(prefix)));
+               changed => !tracerExcludePaths.Any(prefix => changed.StartsWith(prefix)));
+
+           var profilerMatchedPaths = profilerPathMatcher.Match(gitChanges);
+
+           StringBuilder message = new();
+           message.Append("Based on git changes, ");
 
            string variableValue;
            if (willConsolidatedPipelineRun)
            {
-               Logger.Info($"Based on git changes, consolidated pipeline will run. Skipping no-op pipeline");
-               variableValue = "{}";
-               AzurePipelines.Instance.SetVariable("noop_run_skip_stages", "false");
+               if (profilerMatchedPaths.HasMatches)
+               {
+                   message.Append("tracer consolidated pipeline and profiler pipeline will run. Skipping no-op pipeline.");
+                   variableValue = "{}";
+                   AzurePipelines.Instance.SetVariable("noop_run_skip_stages", "false");
+               }
+               else
+               {
+                   var stages = (from jobName in GenerateProfilerJobsName(profilerConfig)
+                                 select new { Name = "skip_profiler_" + jobName, Value = new { StageToSkip = jobName } })
+                     .ToDictionary(x => x.Name, x => x.Value);
+
+                   message.Append($"profiler pipeline will not run. Generating github status updates for {stages.Count} stages");
+                   variableValue = JsonConvert.SerializeObject(stages, Formatting.Indented);
+                   AzurePipelines.Instance.SetVariable("noop_run_skip_stages", "true");
+               }
            }
            else
            {
-               var stages = (from stage in config.Stages
+               var stages = (from stage in tracerConfig.Stages
                              select new { Name = "skip_" + stage.Stage, Value = new { StageToSkip = stage.Stage } })
                   .ToDictionary(x => x.Name, x => x.Value);
 
-               Logger.Info($"Based on git changes, consolidated pipeline will not run. Generating github status updates for {stages.Count} stages");
+               message.Append("tracer consolidated pipeline ");
+               if (!profilerMatchedPaths.HasMatches)
+               {
+                   foreach (var jobName in GenerateProfilerJobsName(profilerConfig))
```

Comment:
Shall we move the calculated stages for both tracer and profiler outside the if/else. We could then _potentially_ simplify the nested if else into a switch?

Reply:
maybe, not sure how simple will it be.